src/features/inbox/data.ts

```diff
@@ -273,21 +248,96 @@ export async function checkAndTriggerOverdueInboxReminders(): Promise<void> {
  * @param type "new" for a newly created entry, or "reminder" for an inbox reminder.
  * @returns The response from slack.
  */
-async function sendInboxNotification(to: string, entry: InboxEntry, type: "new" | "reminder"): Promise<ChatPostMessageResponse> {
-
+async function sendInboxNotification(
+  to: string,
+  entry: InboxEntry,
+  type: "new" | "reminder"
+): Promise<ChatPostMessageResponse> {
   var title = type == "new" ? "New Inbox Message" : "Inbox Reminder";
-  
+
   var note = "";
   if (entry.deadline != null) {
-    var timeLeft = asReadableDuration(Date.now() - new Date(entry.deadline!).valueOf());
-
-    note = type == "new" ? `You have ${timeLeft} to respond` : `You have ${timeLeft} to respond to this message`;
+    var timeLeft = asReadableDuration(
+      new Date(entry.deadline!).valueOf() - Date.now()
+    );
+
+    //TODO: deadline is not correctly inserted based on the test data
+    note =
+      type == "new"
+        ? `You have *${timeLeft}* to respond`
+        : `You have *${timeLeft}* to respond to this message`;
   }
 
   var response = await slack.client.chat.postMessage({
     channel: to,
-    text: `📬 ${title}${note.length > 0 ? ` | ${note}` : ''}:\n${entry.description}`,
-    // TODO: add blocks to make a nice message ui
+    //text is fallback in case client doesn't support blocks
+    text: `📬 ${title}${note.length > 0 ? ` | ${note}` : ""}:\n${
+      entry.description
+    }`,
+    blocks: [
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: `📬 ${title}${note.length > 0 ? ` | ${note}` : ""}:\n`,
+        },
+      },
+      {
+        type: "divider",
+      },
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: entry.description,
+        },
+      },
+      {
+        type: "divider",
+      },
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: "_You can mark this message as resolved by clicking one of the buttons below. The message will be deleted from your inbox once you do._",
+        },
+      },
+      {
+        type: "actions",
+        elements: [
+          {
+            type: "button",
+            text: {
+              emoji: true,
+              type: "plain_text",
+              text: "✅ Done",
+            },
+            action_id: messageDoneAction,
+            value: JSON.stringify({
+              ts: entry.message.ts,
+              senderId: "", //TODO needs to added but no way to get it from the entry right now
```

Comment:
Type the function param as `entry: ReceivedInboxEntry`. Then you can access the `senderId` from it.